cdc/sink/codec/json.go

```diff
@@ -540,70 +507,14 @@ func (d *JSONEventBatchEncoder) EncodeDDLEvent(e *model.DDLEvent) (*MQMessage, e
 
 // Build implements the EventBatchEncoder interface
 func (d *JSONEventBatchEncoder) Build() (mqMessages []*MQMessage) {
-	if d.supportMixedBuild {
-		if d.valueBuf.Len() == 0 {
-			return nil
-		}
-		/* there could be multiple types of event encoded within a single message which means the type is not sure */
-		ret := NewMQMessage(config.ProtocolOpen, d.keyBuf.Bytes(), d.valueBuf.Bytes(), 0, model.MqMessageTypeUnknown, nil, nil)
-		return []*MQMessage{ret}
-	}
-
 	ret := d.messageBuf
 	d.messageBuf = make([]*MQMessage, 0)
 	return ret
 }
 
-// MixedBuild implements the EventBatchEncoder interface
-func (d *JSONEventBatchEncoder) MixedBuild(withVersion bool) []byte {
-	if !d.supportMixedBuild {
-		log.Panic("mixedBuildSupport not enabled!")
-		return nil
-	}
-	keyBytes := d.keyBuf.Bytes()
-	valueBytes := d.valueBuf.Bytes()
-	mixedBytes := make([]byte, len(keyBytes)+len(valueBytes))
-
-	index := uint64(0)
-	keyIndex := uint64(0)
-	valueIndex := uint64(0)
-
-	if withVersion {
-		// the first 8 bytes is the version, we should copy directly
-		// then skip 8 bytes for next round key value parse
-		copy(mixedBytes[:8], keyBytes[:8])
-		index = uint64(8)    // skip version
-		keyIndex = uint64(8) // skip version
-	}
-
-	for {
-		if keyIndex >= uint64(len(keyBytes)) {
-			break
-		}
-		keyLen := binary.BigEndian.Uint64(keyBytes[keyIndex : keyIndex+8])
-		offset := keyLen + 8
-		copy(mixedBytes[index:index+offset], keyBytes[keyIndex:keyIndex+offset])
-		keyIndex += offset
-		index += offset
-
-		valueLen := binary.BigEndian.Uint64(valueBytes[valueIndex : valueIndex+8])
-		offset = valueLen + 8
-		copy(mixedBytes[index:index+offset], valueBytes[valueIndex:valueIndex+offset])
-		valueIndex += offset
-		index += offset
-	}
-	return mixedBytes
-}
-
 // Size implements the EventBatchEncoder interface
 func (d *JSONEventBatchEncoder) Size() int {
-	return d.keyBuf.Len() + d.valueBuf.Len()
-}
-
-// Reset implements the EventBatchEncoder interface
-func (d *JSONEventBatchEncoder) Reset() {
-	d.keyBuf.Reset()
-	d.valueBuf.Reset()
+	return -1
```

Comment:
Is there any side effect to `return -1` here (before it is removed)?

Reply:
No, because the size of our original json protocol is only for mix encoder. So there is no problem to return -1.